smoketest.sh

```diff
@@ -10,16 +10,54 @@ function runCryostat() {
     local grafanaPort="$(xpath -q -e 'project/properties/cryostat.itest.grafana.port/text()' pom.xml)"
     # credentials `user:pass`
     echo "user:d74ff0ee8da3b9806b18c877dbf29bbde50b5bd8e4dad7a3a725000feb82e8f1" > "./conf/cryostat-users.properties"
+
+    if [ "$1" = "postgres" ]; then
+        JDBC_URL="jdbc:postgresql://cryostat:5432/cryostat"
+        JDBC_DRIVER="org.postgresql.Driver"
+        HIBERNATE_DIALECT="org.hibernate.dialect.PostgreSQL95Dialect"
+        JDBC_USERNAME="postgres"
+        JDBC_PASSWORD="abcd1234"
+        HBM2DDL="update"
+    elif [ "$1" = "h2file" ]; then
+        JDBC_URL="jdbc:h2:file:/opt/cryostat.d/conf.d/h2;INIT=create domain if not exists jsonb as other"
+        HBM2DDL="update"
```

Comment:
Could you explain what happens "behind the scenes" when we set `HBM2DDL="update"`? Does this mean any time we make changes to the schema defined in `src/test/resources/postgres/postgre.sql`, the database instance deployed in `smoketest` will be updated accordingly?

Reply:
The `update` value is only passed in here when we're running in `h2:file` mode, not when we're connecting to a Postgres instance. It means that Hibernate will try to update the database schema for the file-based H2 database instance. The schema is generated by Hibernate based on the `persistence.xml` and the various `Properties` values set in the `StorageModule` (which act as overrides on the `persistence.xml` config). Hibernate inspects all the classes named in the config, which right now is only the `PluginInfo` class, and generates a schema from that object definition.

For Postgres, the schema is hand-written in that `postgre.sql` file that you identified, and it is applied "automatically" in the developer testing workflow [here](https://github.com/cryostatio/cryostat/pull/1032/files#diff-cce023726064db3aa55e2ba6abf9be5551fc6ad70919029d36f9a4a3a1c33f9fR58). This isn't a proper database migration script, it's just an initialization really, so for the development workflow there would still likely be a requirement to either manually drop the existing database table(s) or just blow away the storage directory before being able to re-apply any changes to the `postgre.sql`.

Reply:
> The schema is generated by Hibernate based on the persistence.xml and the various Properties values set in the StorageModule (which act as overrides on the persistence.xml config). Hibernate inspects all the classes named in the config, which right now is only the PluginInfo class, and generates a schema from that object definition. If we add a new field to the PluginInfo for example then on the next rebuild/re-run (in h2:file mode both before and after) Hibernate will add a new corresponding column to the table.

Ah, this makes a lot more sense. Thanks!